### PR TITLE
relay: index inreqs by remoteAddr and id

### PR DESCRIPTION
**Pull request by Raynos**
There is an issue in our relay handler implementation
where we were indexing by id rather then by remoteAddr
and id.

r: @kriskowal @shannili

### CODE REVIEW

File: node/relay_handler.js [hunk 1]
```diff
@@ -168,12 +169,29 @@ RelayHandler.prototype.handleRequest = function handleRequest(req, buildRes) {
...
-        delete self.reqs[req.id];
+        delete self.sockets[req.remoteAddr].reqs[req.id];
     }
 };
 
+RelayHandler.prototype.getReq = function getReq(req) {
+    var self = this;
+
+    var socket = self.sockets[req.remoteAddr];
+    if (!socket) {
```
Comment (ShanniLi) [inline node/relay_handler.js]: Doing a set in get. It is perfect for the context being called, but potential an issue for reuse. May be better to put it into set. 

This comment is no blocking.

**Comment by ShanniLi** [comment]
lgtm

**Comment by ShanniLi** [comment]
lgtm. Like this version better. 
Just to make sure, req.connection.guid should never contain '~', right?

**Comment by kriskowal** [comment]
Apart from the mis-named GUID, this is :ok:. Recommending that we just call them connection identifiers, which will be unique for the duration of a process.

**Pull request merged**